junit5-engine/src/main/java/org/junit/gen5/engine/junit5/execution/JUnit5EngineExecutionContext.java

```diff
@@ -49,11 +52,22 @@ public static Builder builder(JUnit5EngineExecutionContext context) {
 		return new Builder(context.state, null);
 	}
 
+	@Override
+	public void registerPublishHandler(PublishHandler handler) {
+		state.publishHandler = handler;
+	}
+
+	public void publishReportEntry(TestDescriptor descriptor, Map<String, String> entry) {
```

Comment:
Alternatively the executionContext could provide a getter for `PublishHandler`. Then `AbstractDescriptorBasedExtensionContext` would only need the publish handler instead of the full engine context.